src/features/animation.ts

```diff
@@ -0,0 +1,171 @@
+/*
+ * Copyright 2019 Google Inc. All Rights Reserved.
+ * Licensed under the Apache License, Version 2.0 (the 'License');
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an 'AS IS' BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {property} from 'lit-element';
+
+import ModelViewerElementBase, {$needsRender, $scene, $tick, $updateSource} from '../model-viewer-base.js';
+import {Constructor} from '../utils.js';
+
+const MILLISECONDS_PER_SECOND = 1000.0
+
+const $updateAnimation = Symbol('updateAnimation');
+const $updateModelLoadsPromise = Symbol('updateModelLoadsPromise');
+const $modelLoads = Symbol('modelLoads');
+const $rejectModelLoads = Symbol('rejectModelLoads');
+
+export const AnimationMixin =
+    (ModelViewerElement: Constructor<ModelViewerElementBase>):
+        Constructor<ModelViewerElementBase> => {
+          class AnimationModelViewerElement extends ModelViewerElement {
+            @property({type: Boolean}) animated: boolean = false;
+            @property({type: Boolean, attribute: 'pause-animation'})
+            pauseAnimation: boolean = false;
```

Comment:
`pause-animation` seems unused/undocumented (other than in the scenario where it's checked with !animated), is this necessary?

Reply:
Instead of `pause-animation`, there does seem like there would be a need to reset back to the default/T-Pose, so there is a difference between 'pausing' the animation (freezing in the middle of an animation) as opposed to 'removing' the animation -- of course, I don't think it necessary to solve this in this PR

Reply:
Yes, it is different from stopping the animation. Pause animation is documented, and will be necessary in order to implement @yuinchien 's mocks to spec.

Reply:
I think the effect you are describing is achieved when setting `animate` to false. Did you have something else in mind?

Reply:
`pause-animation` was not adequately documented, sorry. I added docs for it.

Reply:
@cdata yes, with `pause-animation` and `!animate`, both 'freeze' and 'reset' flows are supported

Reply:
After API refactor, to "pause" you must invoke `pause()` (there is no corresponding attribute/property). And, to "stop" you subsequently set `.currentTime = 0`.